dist/es2015/pager.js

```diff
@@ -0,0 +1,189 @@
+var _dec, _dec2, _class, _desc, _value, _class2, _descriptor, _descriptor2, _descriptor3, _descriptor4, _descriptor5, _descriptor6;
+
+function _initDefineProp(target, property, descriptor, context) {
+  if (!descriptor) return;
+  Object.defineProperty(target, property, {
+    enumerable: descriptor.enumerable,
+    configurable: descriptor.configurable,
+    writable: descriptor.writable,
+    value: descriptor.initializer ? descriptor.initializer.call(context) : void 0
+  });
+}
+
+function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
+  var desc = {};
+  Object['ke' + 'ys'](descriptor).forEach(function (key) {
+    desc[key] = descriptor[key];
+  });
+  desc.enumerable = !!desc.enumerable;
+  desc.configurable = !!desc.configurable;
+
+  if ('value' in desc || desc.initializer) {
+    desc.writable = true;
+  }
+
+  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
+    return decorator(target, property, desc) || desc;
+  }, desc);
+
+  if (context && desc.initializer !== void 0) {
+    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
+    desc.initializer = undefined;
+  }
+
+  if (desc.initializer === void 0) {
+    Object['define' + 'Property'](target, property, desc);
+    desc = null;
+  }
+
+  return desc;
+}
+
+function _initializerWarningHelper(descriptor, context) {
+  throw new Error('Decorating class property failed. Please ensure that transform-class-properties is enabled.');
+}
+
+import { bindable, customElement, bindingMode } from 'aurelia-framework';
+
+export let Pager = (_dec = customElement('pager'), _dec2 = bindable({ defaultBindingMode: bindingMode.twoWay }), _dec(_class = (_class2 = class Pager {
+  constructor() {
+    _initDefineProp(this, 'page', _descriptor, this);
+
+    _initDefineProp(this, 'pagerange', _descriptor2, this);
+
+    _initDefineProp(this, 'limit', _descriptor3, this);
+
+    _initDefineProp(this, 'criteria', _descriptor4, this);
+
+    _initDefineProp(this, 'resource', _descriptor5, this);
+
+    _initDefineProp(this, 'pages', _descriptor6, this);
+  }
+
+  attached() {
+    if (!this.page) {
+      this.page = 1;
+    }
+
+    this.reloadCount();
+  }
+
+  reloadCount() {
+    if (this.resource) {
+      return this._calculatePages();
+    }
+
+    this._calculateRange();
+  }
+
+  pageChanged(newValue, oldValue) {
+    if (newValue !== oldValue) {
+      this.goToPage(newValue);
+    }
+  }
+
+  criteriaChanged(newValue, oldValue) {
```

Comment:
Aurelia observes the properties on a custom element's class instance. This will only work if the criteria object is changed in the following fashion.
```javascript
/* works */
this.criteria.foo = 'bar';
this.criteria = this.criteria;

/* does not work */
this.criteria.foo = 'bar';
```